templates/kubernetes-cluster.template

```diff
@@ -381,6 +383,36 @@ Resources:
             # so pods can communicate
             kubectl apply -f https://${QSS3BucketName}.s3.amazonaws.com/${QSS3KeyPrefix}/scripts/${NetworkingProvider}.yaml
 
+            # Expose the api server over ELB
+            kubectl -n kube-system expose pod \
+              kube-apiserver-$(hostname -f) \
+              --type=LoadBalancer \
+              --port 443 \
+              --target-port=6443 \
+              --name apiserver-public
+
+            # Wait until it's available
+            set +e
+            while :; do
+              LB_HOSTNAME=$(kubectl -n kube-system get svc apiserver-public -o jsonpath='{.status.loadBalancer.ingress[0].hostname}')
+              if [[ $? -eq 0 ]]; then break; fi
+              sleep 1
+            done
+            set -e
+
+            # Grab the client config for the admin user and leave it in the user's home dir, but reconfigure it with
+            # some sed magic.  What we're doing here:
+            # - Replacing the server with the publicly-visible load balancer address
+            # - Disabling TLS verification (since the hostname is random and not one of the SANs in the server cert)
+            # - Omitting the CA data, since it's not allowed when you're disabling TLS verification
```

Comment:
Sure, we should do that and also modify the "kubectl join command" Output section to have a little comment there too.